config/webpack.js

```diff
@@ -0,0 +1,96 @@
+const { join } = require("path");
+const webpack = require("webpack");
+const ExtractTextPlugin = require("extract-text-webpack-plugin");
+
+const Jarvis = require("../lib/server");
+const pkg = require("../package.json");
+
+const babel = require("./babel");
+const styles = require("./style");
+const uglify = require("./uglify");
+
+const dist = join(__dirname, "../lib");
+
+module.exports = env => {
+  const isProd = env && env.production;
+  
+  // Our style-loader chain
+  const cssGroup = styles(isProd);
+  
+  // Our entry file
+  let entry = './src/index.js';
+
```

Comment:
For readability/simplicity/1 thing at a time, I prefer something like...
```javascript
const baseEntry = './src/index.js';
const hmrEntry = ['webpack-hot-middleware/client path=/__webpack_hmr&timeout=20000', baseEntry];
const entry = isProd ? hmrEntry : baseEntry;

const basePlugins  = [
    new webpack.DefinePlugin({
      "process.env.NODE_ENV": JSON.stringify(isProd ? 'production' : 'development')
    })
];

const prodPlugins = [
      new webpack.optimize.UglifyJsPlugin(uglify),
      new ExtractTextPlugin({
        filename: "style.css",
        allChunks: false
      })
];

const hmrPlugins = [
      new webpack.HotModuleReplacementPlugin(),
      new webpack.NoEmitOnErrorsPlugin(),
      new Jarvis()
];

const plugins = isProd ? [...basePlugins, ...prodPlugins] : [...basePlugins, ...hmrPlugins];

if (isProd) { babel.plugins.push('babel-plugin-transform-react-remove-prop-types'); }

return {
    entry,
    plugins,
    yada
}
```

Reply:
As you can see, it's very murky trying to separate this PR. The entire webpack-configuration has been reworked.